builder/pipeline_test.cc

```diff
@@ -1299,4 +1299,71 @@ TEST(pipeline, parallel_stencils) {
   visualize(viz_dir() + "parallel_stencils.html", p, inputs, outputs, &ctx);
 }
 
+TEST(pipeline, diamond_stencils) {
+  // Make the pipeline
+  node_context ctx;
+
+  auto in = buffer_expr::make(ctx, "in1", sizeof(short), 2);
+  auto intm1 = buffer_expr::make(ctx, "intm1", sizeof(short), 2);
+  auto intm2 = buffer_expr::make(ctx, "intm2", sizeof(short), 2);
+  auto intm3 = buffer_expr::make(ctx, "intm3", sizeof(short), 2);
+  auto intm4 = buffer_expr::make(ctx, "intm4", sizeof(short), 2);
+  auto out = buffer_expr::make(ctx, "out", sizeof(short), 2);
+
+  var x(ctx, "x");
+  var y(ctx, "y");
+
+  func add1 = func::make(add_1<short>, {{in, {point(x), point(y)}}}, {{intm1, {x, y}}});
+  func add2 = func::make(multiply_2<short>, {{in, {point(x), point(y)}}}, {{intm2, {x, y}}});
+  func stencil1 = func::make(sum3x3<short>, {{intm1, {bounds(-1, 1) + x, bounds(-1, 1) + y}}}, {{intm3, {x, y}}});
+  func stencil2 = func::make(sum5x5<short>, {{intm2, {bounds(-2, 2) + x, bounds(-2, 2) + y}}}, {{intm4, {x, y}}});
+  func diff = func::make(subtract<short>, {{intm3, {point(x), point(y)}}, {intm4, {point(x), point(y)}}}, {{out, {x, y}}});
+
+  diff.loops({{y, 1}});
+
+  pipeline p = build_pipeline(ctx, {in}, {out});
+
+  // Run the pipeline.
+  const int W = 20;
+  const int H = 10;
+  buffer<short, 2> in_buf({W + 4, H + 4});
+  in_buf.translate(-2, -2);
+  buffer<short, 2> out_buf({W, H});
+
+  init_random(in_buf);
+  out_buf.allocate();
+
+  // Not having span(std::initializer_list<T>) is unfortunate.
```

Comment:
No need to do anything, I just think it's funny how many times I (and now you) have copy-pasted this comment.